diff -up ./sensu.gemspec.bak ./sensu.gemspec
--- ./sensu.gemspec.bak	2015-02-02 09:22:43.392759979 +1000
+++ ./sensu.gemspec	2015-02-02 09:25:38.199688493 +1000
@@ -23,53 +23,53 @@ Gem::Specification.new do |s|
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
       s.add_runtime_dependency(%q<multi_json>, ["= 1.10.1"])
-      s.add_runtime_dependency(%q<uuidtools>, ["= 2.1.4"])
+      s.add_runtime_dependency(%q<uuidtools>, [">= 2.1.4"])
       s.add_runtime_dependency(%q<sensu-em>, ["= 2.4.0"])
       s.add_runtime_dependency(%q<sensu-logger>, ["= 1.0.0"])
       s.add_runtime_dependency(%q<sensu-settings>, ["= 1.2.0"])
       s.add_runtime_dependency(%q<sensu-extension>, ["= 1.0.0"])
       s.add_runtime_dependency(%q<sensu-extensions>, ["= 1.0.0"])
       s.add_runtime_dependency(%q<sensu-transport>, ["= 2.4.0"])
       s.add_runtime_dependency(%q<sensu-spawn>, ["= 1.1.0"])
       s.add_runtime_dependency(%q<em-redis-unified>, ["= 0.5.0"])
-      s.add_runtime_dependency(%q<sinatra>, ["= 1.3.5"])
-      s.add_runtime_dependency(%q<async_sinatra>, ["= 1.0.0"])
-      s.add_runtime_dependency(%q<thin>, ["= 1.5.0"])
+      s.add_runtime_dependency(%q<sinatra>, [">= 1.3.5"])
+      s.add_runtime_dependency(%q<async_sinatra>, [">= 1.0.0"])
+      s.add_runtime_dependency(%q<thin>, [">= 1.5.0"])
       s.add_development_dependency(%q<rake>, ["~> 10.3"])
       s.add_development_dependency(%q<rspec>, ["~> 3.0.0"])
       s.add_development_dependency(%q<em-http-request>, ["~> 1.1"])
     else
       s.add_dependency(%q<multi_json>, ["= 1.10.1"])
-      s.add_dependency(%q<uuidtools>, ["= 2.1.4"])
+      s.add_dependency(%q<uuidtools>, [">= 2.1.4"])
       s.add_dependency(%q<sensu-em>, ["= 2.4.0"])
       s.add_dependency(%q<sensu-logger>, ["= 1.0.0"])
       s.add_dependency(%q<sensu-settings>, ["= 1.2.0"])
       s.add_dependency(%q<sensu-extension>, ["= 1.0.0"])
       s.add_dependency(%q<sensu-extensions>, ["= 1.0.0"])
       s.add_dependency(%q<sensu-transport>, ["= 2.4.0"])
       s.add_dependency(%q<sensu-spawn>, ["= 1.1.0"])
       s.add_dependency(%q<em-redis-unified>, ["= 0.5.0"])
-      s.add_dependency(%q<sinatra>, ["= 1.3.5"])
-      s.add_dependency(%q<async_sinatra>, ["= 1.0.0"])
-      s.add_dependency(%q<thin>, ["= 1.5.0"])
+      s.add_dependency(%q<sinatra>, [">= 1.3.5"])
+      s.add_dependency(%q<async_sinatra>, [">= 1.0.0"])
+      s.add_dependency(%q<thin>, [">= 1.5.0"])
       s.add_dependency(%q<rake>, ["~> 10.3"])
       s.add_dependency(%q<rspec>, ["~> 3.0.0"])
       s.add_dependency(%q<em-http-request>, ["~> 1.1"])
     end
   else
     s.add_dependency(%q<multi_json>, ["= 1.10.1"])
-    s.add_dependency(%q<uuidtools>, ["= 2.1.4"])
+    s.add_dependency(%q<uuidtools>, [">= 2.1.4"])
     s.add_dependency(%q<sensu-em>, ["= 2.4.0"])
     s.add_dependency(%q<sensu-logger>, ["= 1.0.0"])
     s.add_dependency(%q<sensu-settings>, ["= 1.2.0"])
     s.add_dependency(%q<sensu-extension>, ["= 1.0.0"])
     s.add_dependency(%q<sensu-extensions>, ["= 1.0.0"])
     s.add_dependency(%q<sensu-transport>, ["= 2.4.0"])
     s.add_dependency(%q<sensu-spawn>, ["= 1.1.0"])
     s.add_dependency(%q<em-redis-unified>, ["= 0.5.0"])
-    s.add_dependency(%q<sinatra>, ["= 1.3.5"])
-    s.add_dependency(%q<async_sinatra>, ["= 1.0.0"])
-    s.add_dependency(%q<thin>, ["= 1.5.0"])
+    s.add_dependency(%q<sinatra>, [">= 1.3.5"])
+    s.add_dependency(%q<async_sinatra>, [">= 1.0.0"])
+    s.add_dependency(%q<thin>, [">= 1.5.0"])
     s.add_dependency(%q<rake>, ["~> 10.3"])
     s.add_dependency(%q<rspec>, ["~> 3.0.0"])
     s.add_dependency(%q<em-http-request>, ["~> 1.1"])

diff -up ./lib/sensu/utilities.rb.bak ./lib/sensu/utilities.rb
--- ./lib/sensu/utilities.rb.bak	2015-02-02 10:31:09.059329773 +1000
+++ ./lib/sensu/utilities.rb	2015-02-02 10:31:13.841356130 +1000
@@ -1,4 +1,4 @@
-gem 'uuidtools', '2.1.4'
+gem 'uuidtools', '2.1.5'
 
 require 'uuidtools'

diff -up ./lib/sensu/api.rb.bak ./lib/sensu/api.rb
--- ./lib/sensu/api.rb.bak	2015-02-02 11:02:22.985820558 +1000
+++ ./lib/sensu/api.rb	2015-02-02 11:03:23.457161952 +1000
@@ -1,10 +1,10 @@
 require 'sensu/daemon'
 
-gem 'sinatra', '1.3.5'
-gem 'async_sinatra', '1.0.0'
+gem 'sinatra', '>= 1.3.5'
+gem 'async_sinatra', '1.1.0'
 
 unless RUBY_PLATFORM =~ /java/
-  gem 'thin', '1.5.0'
+  gem 'thin', '1.6.2'
   require 'thin'
 end
