From cf80c6dc8f8f70d542d140703e49b704e718e685 Mon Sep 17 00:00:00 2001
From: Jamie Nguyen <j@jamielinux.com>
Date: Tue, 30 Sep 2014 11:46:14 +0100
Subject: [PATCH] Disable internal log rotation

---
 webapp/graphite/logger.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/webapp/graphite/logger.py b/webapp/graphite/logger.py
index afa43f4..d617520 100644
--- a/webapp/graphite/logger.py
+++ b/webapp/graphite/logger.py
@@ -13,7 +13,7 @@ See the License for the specific language governing permissions and
 limitations under the License."""
 
 import os, logging
-from logging.handlers import TimedRotatingFileHandler as Rotater
+from logging import FileHandler
 from django.conf import settings
 
 logging.addLevelName(30,"rendering")
@@ -37,22 +37,22 @@ class GraphiteLogger:
     self.metricAccessLogger = logging.getLogger("metric_access")
     #Setup formatter & handlers
     self.formatter = logging.Formatter("%(asctime)s :: %(message)s","%a %b %d %H:%M:%S %Y")
-    self.infoHandler = Rotater(self.infoLogFile,when="midnight",backupCount=1)
+    self.infoHandler = FileHandler(self.infoLogFile)
     self.infoHandler.setFormatter(self.formatter)
     self.infoLogger.addHandler(self.infoHandler)
-    self.exceptionHandler = Rotater(self.exceptionLogFile,when="midnight",backupCount=1)
+    self.exceptionHandler = FileHandler(self.exceptionLogFile)
     self.exceptionHandler.setFormatter(self.formatter)
     self.exceptionLogger.addHandler(self.exceptionHandler)
     if settings.LOG_CACHE_PERFORMANCE:
-      self.cacheHandler = Rotater(self.cacheLogFile,when="midnight",backupCount=1)
+      self.cacheHandler = FileHandler(self.cacheLogFile)
       self.cacheHandler.setFormatter(self.formatter)
       self.cacheLogger.addHandler(self.cacheHandler)
     if settings.LOG_RENDERING_PERFORMANCE:
-      self.renderingHandler = Rotater(self.renderingLogFile,when="midnight",backupCount=1)
+      self.renderingHandler = FileHandler(self.renderingLogFile)
       self.renderingHandler.setFormatter(self.formatter)
       self.renderingLogger.addHandler(self.renderingHandler)
     if settings.LOG_METRIC_ACCESS:
-      self.metricAccessHandler = Rotater(self.metricAccessLogFile,when="midnight",backupCount=10)
+      self.metricAccessHandler = FileHandler(self.metricAccessLogFile)
       self.metricAccessHandler.setFormatter(self.formatter)
       self.metricAccessLogger.addHandler(self.metricAccessHandler)
 
-- 
1.9.3

